src/Microsoft.DotNet.ProjectModel/DependencyContextBuilder.cs

```diff
@@ -168,10 +168,28 @@ private static CompilationOptions GetCompilationOptions(CommonCompilerOptions co
                     export.Library.Hash,
                     assemblies,
                     libraryDependencies,
-                    serviceable);
+                    serviceable,
+                    GetLibraryPath(export.Library));
             }
         }
 
+        private string GetLibraryPath(LibraryDescription description)
+        {
+            var packageDescription = description as PackageDescription;
+
+            if (packageDescription != null)
+            {
+                // This is the relative path appended to a NuGet packages directory to find the directory containing
+                // the package assets. This string should mastered only byNuGet, but has the format:
+                // {lowercase-package-ID}/{lowercase-package-version}
+                // 
+                // For example: newtonsoft.json/9.0.1
+                return packageDescription.PackageLibrary?.Path;
```

Comment:
I was thinking that visual example would help people understand how this data flows from the lock file, but I can certainly remove it (since NuGet masters this... it's possible that this comment will go out of date and will be more confusing that helpful...).